src/tasks/sentry/tasksentry.cpp

```diff
@@ -32,7 +75,68 @@ void TaskSentry::run() {
   connect(request, &NetworkRequest::requestCompleted, this,
           [this](const QByteArray& data) {
             Q_UNUSED(data);
-            logger.debug() << "Sentry sent events";
+            // Let's note the event id in the logs, so we can
+            // connect customer support logs with sentry :)
+            logger.debug() << "Sentry sent event (" << m_eventID << ")";
             emit completed();
           });
-}
\ No newline at end of file
+}
+
+bool TaskSentry::isCrashReport() {
+  if (m_checkedContent) {
+    // We already checked this envelope, no need to do that twice.
+    return m_isCrashReport;
+  }
+  parseEnvelope();
+  m_checkedContent = true;
+  return m_isCrashReport;
+}
+
+void TaskSentry::parseEnvelope() {
+  auto envelope = QString::fromUtf8(m_envelope);
+
+  // The Sentry Envelope is a basic format
+  // It is a series of JSON objects splited by lines.
+  QStringList objects = envelope.split("\n");
+  if (objects.empty()) {
+    Q_UNREACHABLE();  // This really should not happen.
+  }
+  // The fist line is always the header:
+  // a json object, containing the DSN and an event ID
+  auto header = objects.takeFirst();
+  if (!header.contains("\"event_id\":")) {
```

Comment:
Please, parse it as JSON:
- if it's not an object
- or the object does not contain event_id...